src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  useGetBackersRewardPercentage,
+  useGetRewardsCoinbase,
+  useGetRewardsERC20,
+} from '@/app/collective-rewards/rewards'
+import { usePricesContext } from '@/shared/context/PricesContext'
+import { useGetBuildersByState } from '@/app/collective-rewards/user'
+import { RequiredBuilder } from '@/app/collective-rewards/types'
+import { useGaugesGetFunction } from '@/app/collective-rewards/shared'
+import { useCycleContext } from '@/app/collective-rewards/metrics'
+import { formatEther } from 'viem'
+import { useMemo } from 'react'
+
+export const useGetABI = () => {
+  const {
+    data: builders,
+    isLoading: buildersLoading,
+    error: buildersError,
+  } = useGetBuildersByState<RequiredBuilder>({
+    activated: true,
+    communityApproved: true,
+    kycApproved: true,
+    revoked: false,
+  })
+
+  const { data: rifRewards, isLoading: rifRewardsLoading, error: rifRewardsError } = useGetRewardsERC20()
+  const {
+    data: rbtcRewards,
+    isLoading: rbtcRewardsLoading,
+    error: rbtcRewardsError,
+  } = useGetRewardsCoinbase()
+
+  const {
+    data: { cycleNext },
+    isLoading: cycleLoading,
+    error: cycleError,
+  } = useCycleContext()
+
+  const gauges = builders.map(({ gauge }) => gauge)
+  const {
+    data: totalAllocation,
+    isLoading: totalAllocationLoading,
+    error: totalAllocationError,
+  } = useGaugesGetFunction(gauges, 'totalAllocation')
+
+  const buildersAddress = builders.map(({ address }) => address)
+  const {
+    data: backersRewardsPct,
+    isLoading: backersRewardsPctLoading,
+    error: backersRewardsPctError,
+  } = useGetBackersRewardPercentage(buildersAddress, cycleNext.toSeconds())
+
+  const { prices } = usePricesContext()
+
+  const abi = useMemo(() => {
+    const sumTotalAllocation = Object.values(totalAllocation).reduce((acc, value) => acc + (value ?? 0n), 0n)
+
+    if (!sumTotalAllocation) {
+      return 0
+    }
+
+    const rifPrice = prices.RIF?.price ?? 0
+    const rbtcPrice = prices.RBTC?.price ?? 0
+    const rifAmount = Number(formatEther(rifRewards ?? 0n))
+    const rbtcAmount = Number(formatEther(rbtcRewards ?? 0n))
+    const cyclePayout = rifAmount * rifPrice + rbtcAmount * rbtcPrice
```

Comment:
Im not sure if I should extract this into a hook or function, if I do it we will introduce another useMemo

Reply:
I don't think we need it for now 👍🏼